### PR TITLE
Rewrite frontpage to use react

### PR DESCRIPTION
**Pull request by yuvipanda**
Currently, the front page is a jinja2 rendered template,
served by JupyterHub. This PR replaces it with a react app
using Chakra UI. This gives us the following advantages:

- It reads config from a different YAML file now, but should be
  eventually moved to read the config from jupyterhub-configurator!
  This gives full control of the front page to hub admins, reducing
  workload on 2i2c engineers. 
- Allows for client-side markdown rendering, so config fields for
  users can be more granular - 'tagline' instead of separate
  'funded by', 'operated by', 'designed by' variables.
- A fully static deployment of this is available on netlify,
  at
  https://deploy-preview-4--festive-khorana-08aa29.netlify.app/?host=staging.pilot.2i2c.cloud.
  You can change the value of the host= parameter in the URL to
  see how this would look for different hubs.
- Opens up possibilities for future customization - for example,
  adding nbgitpuller functionality, or easy way to display 'notices'

One major disadvantage is that the bundle size is pretty big now
(~300KB!), but that should come down significantly once
https://github.com/chakra-ui/chakra-ui/pull/3488 is merged

TODO:

- [x] Convert all existing hub config from pilot-hubs repo to
       YAML file here
- [ ] Add more docs on local testing
- [x] Split index.js into separate files
- [ ] Figure out deployment workflow for deploying to hubs
- [x] Make JSON Schema for hubs.yaml file
- [x] Re-add logic for sending users to appropriate UI after logging in

### CODE REVIEW

**Comment by yuvipanda** [comment]
See the deploy preview here: https://deploy-preview-4--festive-khorana-08aa29.netlify.app/?host=staging.pilot.2i2c.cloud

**Comment by damianavila** [comment]
I am not a React guy, so I do not follow React-related trends... is Chakra (I like that name!) a popular (hot) framework in the React world lately? Just curious :wink:

**Comment by yuvipanda** [comment]
@damianavila yeah, I think it's the second most popular framework (after material UI, not counting bootstrap), so seems to have good amount of community support.

File: src/hubs.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+vars:
+  2i2cSubTitle: &2i2cSubTitle |
+      This is a pilot service running on open source infrastructure. See the
+      [2i2c Pilot documentation](https://pilot.2i2c.org) for usage and
+      deployment information.
+  2i2cTagline: &2i2cTagline |
+    Designed, Operated and Funded by [2i2c.org](https://2i2c.org)
+  2i2cLogo: &2i2cLogo
+      img: https://2i2c.org/media/logo.png
+      width: 480px
```
Comment (damianavila) [inline src/hubs.yaml]: OK... all of this is really nice!!
Super easy to customize for admins!

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  const freshPage = nextUrl === null || nextUrl === '' || nextUrl === '/hub/';
+  console.log(urlParams.get('next'))
+
+  return <Flex alignItems="center" direction="column" {...props}>
+    {freshPage &&
+      <RadioGroup defaultValue="notebook" paddingBottom={4}>
+        <Stack spacing={4} direction="row">
+          <Text>After logging in, open</Text>
+          <Radio value="notebook">Jupyter Notebook</Radio>
+          <Radio value="rstudio">RStudio</Radio>
```
Comment (damianavila) [inline src/index.js]: Any reason to have RStudio as the second option? I would expect Lab before RStudio unless you want to five R stuff more prominence...

Comment (yuvipanda) [reply src/index.js]: @damianavila ah, was just reproducing what's there now. Also breaks up the Jupyter monotony :D

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      dangerouslySetInnerHTML={{ __html: md.render(tagline) }} />
+  </Flex>
+}
+
+const getCurrentHub = () => {
+  const defaults = hubsConfig.defaults;
+  const queryParams = new URLSearchParams(location.search);
+  // FIXME: This should only work with dev builds
+  // Can be a security issue in prod builds, since one hub
+  // can be made to look like a different hub
```
Comment (damianavila) [inline src/index.js]: I think this is OK for exploratory work happening in this PR... but I would be -1 about providing this parameter injection tool :wink: in any released stuff.

Comment (yuvipanda) [reply src/index.js]: @damianavila yeah, totally :D I'll put a dev mode guard around this.

Comment (yuvipanda) [reply src/index.js]: @damianavila so i couldn't figure out how to tell webpack to only include things in dev mode, so I've just added more guards instead here. Not sure that's sufficient though.

Comment (damianavila) [reply src/index.js]: I will take a look!

File: build.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import os
+import shutil
+import subprocess
+import jsonschema
+from ruamel.yaml import YAML
+
+yaml = YAML(typ="safe")
+
+
+def validate():
```
Comment (damianavila) [inline build.py:11]: Super! Yesterday I was thinking about validating the admin's stuff somehow, glad to see this piece here, that should prevent us/them a lot of errors.

File: src/loginpage.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  })
+
+  return hub;
+}
+
+const HubChooser = ({ setHub }) => {
+  const domain = location.hostname;
+  const queryDomain = new URLSearchParams(location.search).get('domain');
+  if (queryDomain !== null) {
+    if (domain !== 'localhost' && !domain.endsWith('.netlify.app')) {
```
Comment (damianavila) [inline src/loginpage.js:40]: Even when the attack surface has diminished, I think this is still problematic...

Maybe you can use the webpack "mode" to only add this piece in dev mode?

https://webpack.js.org/guides/production/#specify-the-mode

And have something like this...

```javascript
if (process.env.NODE_ENV !== 'production') {
  Yuvi injection tool here... ;-)
}
```

**Comment by yuvipanda** [comment]
Unfortunately, this was a dead end. Closing.

**Pull request closed without merging**